src/main/java/uk/gov/justice/digital/hmpps/csr/api/repository/SqlRepository.kt

```diff
@@ -278,14 +278,14 @@ class SqlRepository(private val jdbcTemplate: NamedParameterJdbcTemplate) {
                 n.on_date,
                 usr.name AS quantumId, 
                 CASE WHEN n.action_type is null
-                then (
+                then coalesce(
                       SELECT MAX(tw_protocol.lastmodified)
                       FROM tw_protocol
                       WHERE st_staff_id = n.st_staff_id
                       AND level_id = n.level_id
                       AND layer = -1 
                       AND tw_protocol.on_date BETWEEN (SYSDATE - 1) AND (SYSDATE + 2)
-                     ) 
+                     , n.lastmodified) -- backup if no tw_protocol row
```

Comment:
where a row exists in TW_PROTOCOL it will be more recent (possibly an edit) than the lastmodified value in CMD_NOTIFICATION, but if not then the CMD_NOTIFICATION is when the shift was added.